srml/system/src/lib.rs

```diff
@@ -378,6 +401,54 @@ pub fn ensure_none<OuterOrigin, AccountId>(o: OuterOrigin) -> Result<(), &'stati
 }
 
 impl<T: Trait> Module<T> {
+	/// Deposits an event into this block's event record adding this event
+	/// to the corresponding topic indexes.
+	///
+	/// This will update storage entries that correpond to the specified topics.
+	/// It is expected that light-clients could subscribe to this topics.
+	pub fn deposit_event_indexed(topics: &[T::Hash], event: T::Event) {
+		let extrinsic_index = Self::extrinsic_index();
+		let phase = extrinsic_index.map_or(Phase::Finalization, |c| Phase::ApplyExtrinsic(c));
+		let event = EventRecord {
+			phase,
+			event,
+			topics: topics.iter().cloned().collect::<Vec<_>>(),
+		};
+
+		// Index of the to be added event.
+		let event_idx = {
+			let old_event_count = <EventCount<T>>::get();
+			let new_event_count = match old_event_count.checked_add(1) {
+				// We've reached the maximum number of events at this block, just
```

Comment:
This means that event delivery is unreliable, which could lead to starvation or deadlock in some protocols.

Reply:
Ah, this is more related to #2491 or even before that, to [#2282](https://github.com/paritytech/substrate/commit/89f951f4bff3755d02201c63e08a7d0a05a1b721#diff-24bd3cac4cae1c620f82f7aa79ef28ceR185). 

I believe this code is a mere defending practice and will not be triggered. Here is my logic: wasm32 is a 32-bit platform. It can have at most 4GB of memory, that is, 2^32 bytes. Considering an element to be `u8`, by the time we reach this number of elements the memory will already be full.

If you are still concerned by this issue, I think it would be best to move this discussion into a new issue (or at least to the now closed #2282)